src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -132,9 +132,16 @@ private static bool IsTls13Disabled()
             {
                 using var regKey = Registry.LocalMachine.OpenSubKey(key);
 
-                if (regKey is null) return false;
+                if (regKey is null) continue;
 
-                if (regKey.GetValue("Enabled") is int enabled && enabled == 0)
+                if (regKey.GetValue("Enabled") is int enabled)
+                {
+                    if (enabled == 0) return true;
+                    else continue;
+                }
+
+                // if Enabled key is not present, check the default settings
```

Comment:
Do they get returned sorted? Or how do you know that by this point there isn't `Enable` present?
If they cannot be both present going against each other, that's fine, but the comment is slightly misleading.
I don't know how exactly the registry API behaves, so sorry in advance if my question doesn't make sense.

Reply:
> Or how do you know that by this point there isn't Enable present?

We are checking it 4 lines above :D or I don't understand the question.

Unfortunately, it seems that `Enabled` and `DisabledByDefault` are not redundant, some info can be found [at StackOverflow](https://stackoverflow.com/a/51962330). From my understanding of it, the `Enabled` key takes precedence.

Reply:
I have taken another look into this, and it seems that _either_ of these can cause the setup to fail, so I will need to improve it further